### PR TITLE
Bugfix - Non logging waf_v2 rules

### LINKED ISSUE #6: firehose_arn must be specified in module "fms"
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[Slack Community](https://slack.cloudposse.com)

## Describe the Bug
firehose_arn must be specified in module "fms".
if the variable is not specified terraform complains on
```
logging_configuration = local.enabled && var.firehose_enabled ? local.logging_config_firehose_enabled : local.enabled && var.firehose_arn != null ? local.logging_config_firehose_arn : {}
```
line.
I went ahead and removed logging_configuration to make the module working in my fork.
Then aws_fms_policy resource complained that firehose_arn has to be present.
When I specified firehose_arn attribute, it worked.

## Expected Behavior
The action item here is to update the example and make it clear for the user that firehose_arn attribute is required.
Also, there is an opportunity to update logic in main.tf to remove the nullability check for the variable. (nice to have, not required)

### PR DESCRIPTION
**Pull request by Benbentwo**
## what
* Fixes Non logging configurations for waf_v2 rulesets.
* Fixes locals ternary operator by using json encode to convert types to string.
* Adds me to contributors

## why
* Bugs prevented plan and apply by object object type difference
* Bugs prevented applying rules that did not include firehose

## references
* closes #6

### CODE REVIEW

**Comment by Benbentwo** [comment]
affects https://github.com/cloudposse/terraform-aws-firewall-manager/blob/master/waf_v2.tf#L38-L54

File: examples/complete/providers.tf [hunk 1]
```diff
@@ -0,0 +1,16 @@
+provider "aws" {
+  region = "us-east-1"
```
Comment (aknysh) [inline examples/complete/providers.tf:2]: we usually provide `region` in a var for the example
(which was deleted in `main.tf` but should be in `variables.tf` and set in TF defaults

**Review by aknysh — COMMENTED** [review]
LGTM, ne nitpick

File: examples/complete/main.tf [hunk 2]
```diff
@@ -11,8 +11,26 @@ module "vpc" {
...
-  source = "../.."
+module "firewall_manager" {
+  source = "../../"
+
+  providers = {
+    aws.admin = aws.admin
+    aws       = aws
+  }
+
+  context = module.this.context
```
Comment (aknysh) [inline examples/complete/main.tf:22]: please remove the extra empty lines and move `context = module.this.context` to the last line of the module invocation (per our standard practices)

File: examples/complete/main.tf [hunk 3]
```diff
@@ -1,5 +1,5 @@
-provider "aws" {
-  region = var.region
+locals {
+  assumed_arn = var.is_destroy ? var.firewall_manager_administrator_role_arn : var.organization_management_role_arn
```
Comment (aknysh) [inline examples/complete/main.tf:2]:
```suggestion
  assume_role_arn = var.is_destroy ? var.firewall_manager_administrator_role_arn : var.organization_management_role_arn
```

File: examples/complete/providers.tf [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  region = var.region
+  assume_role {
+    role_arn = var.firewall_manager_administrator_role_arn
+  }
+}
+
+provider "aws" {
+  region = var.region
+  assume_role {
+    role_arn = local.assumed_arn
```
Comment (aknysh) [inline examples/complete/providers.tf:11]:
```suggestion
    role_arn = local.assume_role_arn
```

File: main.tf [hunk 5]
```diff
@@ -10,10 +10,10 @@ locals {
   network_firewall_policies              = local.enabled && length(var.network_firewall_policies) > 0 ? { for policy in flatten(var.network_firewall_policies) : policy.name => policy } : {}
 
 
-  logging_config_firehose_arn     = { logDestinationConfigs : [var.firehose_arn], redactedFields : [{ redactedFieldType : "SingleHeader", redactedFieldValue : "Cookies" }, { redactedFieldType : "Method" }] }
-  logging_config_firehose_enabled = { logDestinationConfigs : [join("", aws_kinesis_firehose_delivery_stream.firehose_stream.*.id)], redactedFields : [{ redactedFieldType : "SingleHeader", redactedFieldValue : "Cookies" }, { redactedFieldType : "Method" }] }
+  logging_config_firehose_arn     = jsonencode({ logDestinationConfigs : [var.firehose_arn], redactedFields : [{ redactedFieldType : "SingleHeader", redactedFieldValue : "Cookies" }, { redactedFieldType : "Method" }] })
```
Comment (aknysh) [inline main.tf:13]: do we need to `jsonencode` it here?
we already `jsonencode` the whole thing https://github.com/cloudposse/terraform-aws-firewall-manager/blob/master/waf_v2.tf#L41

**Review by aknysh — COMMENTED** [review]
@Benbentwo please see comments

File: main.tf (hunk 5 above)
Comment (Benbentwo) [reply main.tf:13]: We need the jsonencode for the ternary condition to have the same object types. TF unfortunately doesn't allow different typed objects in the conditional statement, even if its object object (with different number of fields. which is the case here from 2 fields `logDestinationConfigs` and `redactedFields` vs `{}`)

**Review by max-lobur — APPROVED** [review]

File: main.tf (hunk 5 above)
Comment (dlyubarskiy) [reply main.tf:13]: @Benbentwo I believe this change results in a mis-formatted `managedServiceData` block. There're too many backslashes, around `loggingConfiguration` elements, it looks like this:

\"loggingConfiguration\":\"{\\\"logDestinationConfigs\\\":[\\\"<stream_arn>\\\"],\\\"redactedFields\\\":[{\\\"redactedFieldType\\\":\\\"SingleHeader\\\",\\\"redactedFieldValue\\\":\\\"Cookies\\\"}

and gets rejected by AWS API. This is with TF 1.0.9 and AWS Provider v3.73.0. Removing `jsonencode` solved the issue.

**Pull request merged**